src/sinks/splunk_hec/logs/sink.rs

```diff
@@ -131,54 +134,53 @@ impl ByteSizeOf for HecLogsProcessedEventMetadata {
 
 pub type HecProcessedEvent = ProcessedEvent<LogEvent, HecLogsProcessedEventMetadata>;
 
-pub fn process_log(
-    event: Event,
-    sourcetype: Option<&Template>,
-    source: Option<&Template>,
-    index: Option<&Template>,
-    host_key: &str,
-    indexed_fields: &[String],
-    timestamp_nanos_key: Option<&str>,
-    timestamp_key: &str,
-) -> HecProcessedEvent {
+pub fn process_log(event: Event, data: &HecLogData) -> HecProcessedEvent {
     let event_byte_size = event.size_of();
     let mut log = event.into_log();
 
-    let sourcetype =
-        sourcetype.and_then(|sourcetype| render_template_string(sourcetype, &log, "sourcetype"));
+    let sourcetype = data
+        .sourcetype
+        .and_then(|sourcetype| render_template_string(sourcetype, &log, "sourcetype"));
 
-    let source = source.and_then(|source| render_template_string(source, &log, "source"));
+    let source = data
+        .source
+        .and_then(|source| render_template_string(source, &log, "source"));
 
-    let index = index.and_then(|index| render_template_string(index, &log, "index"));
+    let index = data
+        .index
+        .and_then(|index| render_template_string(index, &log, "index"));
 
-    let host = log.get(host_key).cloned();
+    let host = log.get(data.host_key).cloned();
 
-    let timestamp = if timestamp_key.is_empty() {
+    let timestamp = if data.timestamp_key.is_empty() {
         None
     } else {
-        match log.remove(timestamp_key) {
+        match log.remove(data.timestamp_key) {
             Some(Value::Timestamp(ts)) => {
                 // set nanos in log if valid timestamp in event and timestamp_nanos_key is configured
-                if let Some(key) = timestamp_nanos_key {
+                if let Some(key) = data.timestamp_nanos_key {
                     log.try_insert(path!(key), ts.timestamp_subsec_nanos() % 1_000_000);
                 }
                 Some((ts.timestamp_millis() as f64) / 1000f64)
             }
             Some(value) => {
-                warn!(
-                    "timestamp_key is not valid timestamp: type is {}. Ignoring and allowing splunk to set timestamp.",
-                    value.kind_str()
-                );
+                emit!(SplunkEventTimestampWarning {
+                    message: &format!("timestamp_key is not valid timestamp: type is {}. Ignoring and allowing splunk to set timestamp.",
+                        value.kind_str())
+                });
                 None
             }
             None => {
-                warn!("No timestamp was not found. Allowing splunk to set timestamp.");
+                emit!(SplunkEventTimestampWarning {
+                    message: "No timestamp was not found. Allowing splunk to set timestamp."
+                });
```

Comment:
Should this be debug since this is the expected behavior - allowing splunk to set the timestamp?

Reply:
Good question, I am not sure.
To clarify, this scenario is where a timestamp_key was present (could be the default) , but no timestamp was detected in the log event (`log.remove()` returned `None`).
As opposed to the scenario where the timestamp_key was set to empty string to intentionally suppress the timestamp.

Reply:
Ah, my misunderstanding 👍